linkerd/tls/src/client.rs

```diff
@@ -1,25 +1,47 @@
-use super::Conditional;
 use futures::{
     future::{Either, MapOk},
     prelude::*,
 };
-use linkerd_identity as identity;
+use linkerd_conditional::Conditional;
+use linkerd_identity as id;
 use linkerd_io as io;
 use linkerd_stack::layer;
-pub use rustls::ClientConfig as Config;
 use std::{
+    fmt,
     future::Future,
     pin::Pin,
+    str::FromStr,
     sync::Arc,
     task::{Context, Poll},
 };
 use tokio_rustls::client::TlsStream;
 use tracing::{debug, trace};
 
-pub trait HasConfig {
-    fn tls_client_config(&self) -> Arc<Config>;
+/// A marker type for target server identities.
+#[derive(Clone, Debug, Eq, PartialEq, Hash)]
+pub struct ServerId(pub id::Name);
```

Comment:
it _feels_ weird elsewhere to see a type named `ServerId` imported from the `client` module, but after actually thinking about it, it _does_ make sense...and I don't think we should give this a wordier name to clarify that. so :+1: